### PR TITLE
Reboot log entry should not be in CBOR when printed on console

### PR DESCRIPTION
**Pull request by vrahane**
This PR also adds a package for a generic logging cbor reader utility which would be used by shell instead of having it's own reader: `sys/log/util/cbor_reader`

This fixes https://github.com/apache/mynewt-core/issues/2079

**Before**
```
compat> reset
000000 [ts=0us, mod=6 level=4 ]?crsndSOFTccnt.cimgg5.0.0.0dhashx@065b72e0c0d9af335d67cface6f7
d9d217aaefddffddde672eb8dd2059721c66eflagsx.active bootable confirmed?
Slinky 5.0.0.0
```
**After**
```
compat> reset
000000 [ts=0us, mod=6 level=4 ]{_ "rsn": "SOFT", "cnt": 267, "img": "0.0.0.0", "hash": "9225ff83589e177fc4b3e483d9a6249866bd65b56e5d87dc7a5b4331f3
e85def", "flags": "active bootable confirmed"}
Slinky 5.0.0.0
```

### CODE REVIEW

File: sys/log/full/src/log_console.c [hunk 1]
```diff
@@ -24,9 +24,28 @@
...
 #include "log/log.h"
+#if MYNEWT_VAL(LOG_VERSION) > 2
+#include "tinycbor/cbor.h"
+#include "tinycbor/cbor_buf_reader.h"
+#endif
 
 static struct log log_console;
 
+static int
+console_log_dump_cbor_entry(struct log *log, const void *dptr, uint16_t len)
```
Comment (ccollins476ad) [inline sys/log/full/src/log_console.c]: You could remove the first parameter to this function (`struct log *log`).

File: sys/log/util/pkg.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
+#
```
Comment (ccollins476ad) [inline sys/log/util/pkg.yml]: Does `sys/log/util` itself need to be a package?  It doesn't seem to contain any code of its own.

File: sys/log/util/cbor_reader/include/cbor_reader.h [hunk 3]
```diff
@@ -0,0 +1,33 @@
+/*
```
Comment (ccollins476ad) [inline sys/log/util/cbor_reader/include/cbor_reader.h]: I think `log_cbor_reader` would be a better name for this directory than `cbor_reader`.  If I see
```
#include "cbor_reader.h"
```

It wouldn't occur to me that the file includes a log header.

Also, I think there should be an extra subdirectory under `include` to maintain consistency, i.e.,
```
 sys/log/util/cbor_reader/include/cbor_reader/...
```
```
#include "cbor_reader/cbor_reader.h"
```

**Comment by ccollins476ad** [comment]
```
000000 [ts=0us, mod=6 level=4 ]{_ "rsn": "SOFT", "cnt": 267, "img": "0.0.0.0", "hash": "9225ff83589e177fc4b3e483d9a6249866bd65b56e5d87dc7a5b4331f3
e85def", "flags": "active bootable confirmed"}
```
I wonder where that underscore before `rsn` comes from.

File: sys/log/full/src/log_console.c (hunk 1 above)
Comment (vrahane) [reply sys/log/full/src/log_console.c]: done

File: sys/log/util/pkg.yml (hunk 2 above)
Comment (vrahane) [reply sys/log/util/pkg.yml]: done

File: sys/log/util/cbor_reader/include/cbor_reader.h (hunk 3 above)
Comment (vrahane) [reply sys/log/util/cbor_reader/include/cbor_reader.h]: done

**Comment by vrahane** [comment]
@ccollins476ad Thanks for the review, fixed the issues.

**Comment by vrahane** [comment]
I spoke to @ccollins476ad but just for the record I'll also mention this here. `cbor_value_to_pretty()` prints an underscore `_` if the length of the value is unknown.

`encoding/tinycbor/src/cborpretty.c`
```
278         if (!cbor_value_is_length_known(it)) {
279             if (fprintf(out, "_ ") < 0)
280                 return CborErrorIO;
281         }
```

**Pull request merged**